tst/com/amazon/ion/benchmark/dataGeneratorTest.java

```diff
@@ -0,0 +1,202 @@
+package com.amazon.ion.benchmark;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.amazon.ion.IonReader;
+import com.amazon.ion.IonType;
+import com.amazon.ion.Timestamp;
+import com.amazon.ion.system.IonReaderBuilder;
+import com.amazon.ion.util.IonStreamUtils;
+import org.junit.*;
+
+import java.io.BufferedInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.math.BigInteger;
+import java.nio.channels.FileChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+public class dataGeneratorTest {
+
+    /**
+     * Construct the input Map <String, Object>, the key is the name of the option, the value is the parameter provided in the unit test.
+     * @param size specifies the size in bytes of the generated file.
+     * @param dataType represents the scalar type of Ion data.
+     * @param textCodePointRange provides the range of Unicode code point of characters which construct the Ion string.
+     * @param coefficientDigit the range of digit number of coefficient when the decimal represented in coefficient * 10 ^ exponent.
+     * @param path the destination of the generated file.
+     * @param timestampsTemplate is a string which provides a series of template timestamps which data generating process will follow with.
+     * @param format the format of output file (ion_binary | ion_text).
+     * @param expRange the range of exponent when the decimal represented in coefficient * 10 ^ exponent.
+     * @return the constructed hashmap which can be used in the following test process.
+     */
+    private static Map<String, Object> testCasesGenerator (Object size, Object dataType, Object textCodePointRange, Object coefficientDigit, Object path, Object timestampsTemplate, Object format, Object expRange) {
+        HashMap <String, Object> optionsMap = new HashMap<String, Object>();
+        optionsMap.put("--data-size", size);
+        optionsMap.put("generate", "true");
+        optionsMap.put("--data-type", dataType);
+        optionsMap.put("--text-code-point-range", textCodePointRange);
+        optionsMap.put("--decimal-coefficient-digit-range", coefficientDigit);
+        optionsMap.put("<output_file>", path);
+        optionsMap.put("--timestamps-template", timestampsTemplate);
+        optionsMap.put("--decimal-exponent-range", expRange);
+        optionsMap.put("--format", format);
+        return  optionsMap;
+    }
+
+    /**
+     * Assert generated Ion data is the same type as expected.
+     * @throws Exception if error occurs when executing Ion data generator.
+     */
+    @Test
+    public void testGeneratedType() throws Exception {
+        Map<String, Object> optionsMap = dataGeneratorTest.testCasesGenerator("500","decimal", "[1,1114111]", "[1,10]", "test1.ion", "2021T", "[ion_text]", "[-13,-2]");
```

Comment:
"test1.ion" (and all of the other tests files generated by tests) need to be deleted up after the test finishes, otherwise it will stay on disk and you may risk appending to it in future tests. Consider using something like an `@After` method to help with this.